packages/plugins/plugin-assistant/src/components/QueryEditor/query-extension.ts

```diff
@@ -0,0 +1,335 @@
+      .child(
+        Domino.of('span')
+          .text('type')
+          .classNames('flex items-center text-xs font-thin pis-1 pie-1 rounded-l-[0.2rem] bg-separator'),
+        Domino.of('span').text(label).classNames('leading-[22px] pis-1 pie-1 pb-[1px] text-green-500'),
+      )
+      .build();
+  }
+}
+
+/**
+ * { type: "value" }
+ */
+class ObjectWidget extends WidgetType {
+  private readonly _entries: [string, any][];
+  private readonly _json: string;
+
+  constructor(private readonly _props: any) {
+    super();
+    this._entries = Object.entries(this._props);
+    this._json = JSON.stringify(this._props);
+  }
+
+  override ignoreEvent() {
+    return false;
+  }
+
+  override eq(other: this) {
+    return this._json === other._json;
+  }
+
+  override toDOM() {
+    return Domino.of('span')
+      .classNames('inline-flex items-stretch border border-separator divide-x divide-separator rounded-sm')
+      .child(
+        ...this._entries.map(([key, value]) =>
+          Domino.of('span').classNames('pis-1 pie-1').child(
+            //
+            Domino.of('span').classNames('text-infoText').text(key),
+            Domino.of('span').classNames('pis-1').text(value),
+          ),
+        ),
+      )
+      .build();
+  }
+}
+
+/**
+ * Symbol
+ */
+class SymbolWidget extends WidgetType {
+  constructor(private readonly _str: string) {
+    super();
+  }
+
+  override eq(other: this) {
+    return this._str === other._str;
+  }
+
+  override toDOM() {
+    return Domino.of('span').text(this._str).build();
+  }
+}
+
+/**
+ * Define syntax highlighting tags for the query language.
+ */
+const queryHighlighting = styleTags({
+  // Keywords
+  'Not And Or': t.keyword,
+  TypeKeyword: t.attributeName,
+
+  // Literals
+  String: t.string,
+  Number: t.number,
+  Boolean: t.bool,
+  Null: t.null,
+
+  // Identifiers
+  Identifier: t.variableName,
+  PropertyPath: t.propertyName,
+  PropertyKey: t.propertyName,
+
+  // Punctuation
+  '{ }': t.brace,
+  '[ ]': t.squareBracket,
+  '( )': t.paren,
+  ':': t.definitionOperator,
+  ',': t.separator,
+  '.': t.derefOperator,
+});
+
+/**
+ * Create the query language with the parser and highlighting.
+ */
+const queryLanguage = LRLanguage.define({
+  parser: QueryDSL.Parser.configure({
+    props: [queryHighlighting],
+    strict: false,
+  }),
+  languageData: {
+    commentTokens: { line: '//' },
+  },
+});
+
+/**
+ * Define a custom highlight style for the query language.
+ */
+const queryHighlightStyle = HighlightStyle.define([
+  { tag: t.keyword, class: 'text-blue-500' },
+  { tag: t.string, class: 'text-orange-500' },
+  { tag: t.number, class: 'text-green-500' },
+  { tag: t.bool, class: 'text-green-500' },
+  { tag: t.null, class: 'text-neutral-500' },
+  { tag: t.attributeName, class: 'text-blue-500' },
+  { tag: t.variableName, class: 'text-teal-500' },
+  { tag: t.propertyName, class: 'text-teal-500' },
```

Comment:
We have semantic tokens for these:
```suggestion
  { tag: t.keyword, class: 'text-blueText' },
  { tag: t.string, class: 'text-orangeText' },
  { tag: t.number, class: 'text-greenText' },
  { tag: t.bool, class: 'text-greenText' },
  { tag: t.null, class: 'text-neutralText' },
  { tag: t.attributeName, class: 'text-blueText' },
  { tag: t.variableName, class: 'text-tealText' },
  { tag: t.propertyName, class: 'text-tealText' },
```